Cython/Compiler/Scanning.py

```diff
@@ -485,6 +493,13 @@ def error(self, message, pos=None, fatal=True):
         err = error(pos, message)
         if fatal: raise err
 
+    def error_in_action(self, message, pos=None, fatal=True):
+        # like error, but gets the current scanning position rather than
+        # the position of the last token read
+        if pos is None:
+            pos = (self.name, self.start_line, self.start_col)
+        self.error(message, pos, fatal)
```

Comment:
The comment suggests that this should never be called with a position argument. And scanner errors would always seem fatal.
Also, maybe `error_at_scanpos` might be a more descriptive name than `error_in_action`. And could probably also be added to `Scanning.pxd` as a `cdef` method.
```suggestion
    def error_in_action(self, message):
        # Like error(fatal=True), but gets the current scanning position rather than
        # the position of the last token read.
        pos = (self.name, self.start_line, self.start_col)
        self.error(message, pos)
```